pkg/schedule/schedulers/balance_range.go

```diff
@@ -622,34 +643,26 @@ func (s *balanceRangeScheduler) prepare(cluster sche.SchedulerCluster, opInfluen
 				count += cluster.GetStorePeerCountByRange(source.GetID(), kr.StartKey, kr.EndKey)
 			case core.LearnerScatter:
 				count += cluster.GetStoreLearnerCountByRange(source.GetID(), kr.StartKey, kr.EndKey)
+			default:
+				return nil, errs.ErrInvalidArgument.FastGenByArgs("not supported rule: " + job.Rule.String())
 			}
 		}
 		scoreMap[source.GetID()] = float64(count)
-		totalScore += float64(count)
 	}
 
 	sort.Slice(sources, func(i, j int) bool {
-		rule := job.Rule
-		iop := float64(opInfluence.GetStoreInfluence(sources[i].GetID()).GetStoreInfluenceByRole(rule))
-		jop := float64(opInfluence.GetStoreInfluence(sources[j].GetID()).GetStoreInfluenceByRole(rule))
+		iop := float64(solver.getOpInfluence(sources[i].GetID()))
+		jop := float64(solver.getOpInfluence(sources[j].GetID()))
 		iScore := scoreMap[sources[i].GetID()]
 		jScore := scoreMap[sources[j].GetID()]
 		return iScore+iop > jScore+jop
 	})
 
-	averageScore := float64(0)
-	averageScore = totalScore / float64(len(sources))
-
-	tolerantSizeRatio := int64(float64(totalScore) * adjustRatio)
-	if tolerantSizeRatio < 1 {
-		tolerantSizeRatio = 1
-	}
-	solver.tolerantSource = tolerantSizeRatio
 	return &balanceRangeSchedulerPlan{
```

Comment:
Suggest avoiding using xxxPlan.

Reply:
done， remove the xxplan strcut and put the status to the scheduler